app/scripts/controller-init/bridge-controller-init.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  BridgeClientId,
+  BridgeController,
+  UNIFIED_SWAP_BRIDGE_EVENT_CATEGORY,
+} from '@metamask/bridge-controller';
+import { handleFetch } from '@metamask/controller-utils';
+import { BRIDGE_API_BASE_URL } from '../../../shared/constants/bridge';
+import { trace } from '../../../shared/lib/trace';
+import fetchWithCache from '../../../shared/lib/fetch-with-cache';
+import { getEnvironmentType } from '../lib/util';
+import { ControllerInitFunction } from './types';
+import {
+  BridgeControllerInitMessenger,
+  BridgeControllerMessenger,
+} from './messengers';
+
+type FetchWithCacheOptions = {
+  cacheOptions?: {
+    cacheRefreshTime: number;
+  };
+  functionName?: string;
+};
+
+/**
+ * Initialize the bridge controller.
+ *
+ * @param request - The request object.
+ * @param request.controllerMessenger - The messenger to use for the controller.
+ * @param request.initMessenger - The messenger to use for initialization.
+ * @param request.getController
+ * @returns The initialized controller.
+ */
+export const BridgeControllerInit: ControllerInitFunction<
+  BridgeController,
+  BridgeControllerMessenger,
+  BridgeControllerInitMessenger
+> = ({ controllerMessenger, initMessenger, getController }) => {
+  const transactionController = getController('TransactionController');
+
+  const controller = new BridgeController({
+    messenger: controllerMessenger,
+    clientId: BridgeClientId.EXTENSION,
+
+    // TODO: Remove once TransactionController exports this action type.
+    getLayer1GasFee: (...args) =>
+      transactionController.getLayer1GasFee(...args),
+
+    // @ts-expect-error: Type of `fetchFn` does not match.
+    fetchFn: async (
+      url,
+      { cacheOptions, functionName, ...requestOptions }: FetchWithCacheOptions,
+    ) => {
+      if (functionName === 'fetchBridgeTokens') {
+        return await fetchWithCache({
+          // @ts-expect-error: `url` can be `URL` but `fetchWithCache` expects a
+          // string.
+          url,
+          fetchOptions: { method: 'GET', ...requestOptions },
+          cacheOptions,
+          functionName,
+        });
+      }
```

Comment:
The expected type error can be removed by stringifying the url and adding a fallback for the options

```suggestion

    fetchFn: async (
      url,
      {
        cacheOptions,
        functionName,
        ...requestOptions
      }: FetchWithCacheOptions = {},
    ) => {
      if (functionName === 'fetchBridgeTokens') {
        return await fetchWithCache({
          url: url.toString(),
          fetchOptions: { method: 'GET', ...requestOptions },
          cacheOptions,
          functionName,
        });
      }
```

Reply:
Thanks!